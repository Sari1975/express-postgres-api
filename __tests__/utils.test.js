const {
  generateProductLink,
  generateSingleProductInformation,
} = require('../utils');

test('generateProductLink generates a product link', () => {
  const product = {
    id: 1,
    name: 'MacBook Pro',
    description: 'Laptop from Apple',
    price: '2400.00',
  };
  expect(generateProductLink(product)).toBe(
    `1: <a href="/products?id=1" data-cy="product-link-1">MacBook Pro</a>`,
  );
});

test('generates the single product information', () => {
  const product = {
    id: 3,
    name: 'chair',
    description: 'nice chair',
    price: 33333,
  };

  expect(generateSingleProductInformation(product)).toMatchSnapshot();
});
