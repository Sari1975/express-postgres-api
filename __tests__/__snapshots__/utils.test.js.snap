// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates the single product information 1`] = `"3: chair<br />nice chair<br /><a href=\\"/products\\" data-cy=\\"products-link\\">Back to Products</a>"`;
